src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +241,981 @@ public bool NeedsRefresh()
+                            }
+
+                        case FeatureManagementConstants.Conditions:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                                {
+                                    featureFlag.Conditions = ParseFeatureConditions(ref reader, settingKey);
+                                }
+                                else if (reader.TokenType != JsonTokenType.Null)
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.Conditions,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.StartObject.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.Allocation:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                                {
+                                    featureFlag.Allocation = ParseFeatureAllocation(ref reader, settingKey);
+                                }
+                                else if (reader.TokenType != JsonTokenType.Null)
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.Allocation,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.StartObject.ToString());
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.Variants:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartArray)
+                                {
+                                    List<FeatureVariant> variants = new List<FeatureVariant>();
+
+                                    while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
+                                    {
+                                        if (reader.TokenType == JsonTokenType.StartObject)
+                                        {
+                                            FeatureVariant featureVariant = ParseFeatureVariant(ref reader, settingKey);
+
+                                            if (featureVariant.Name != null)
+                                            {
+                                                variants.Add(featureVariant);
+                                            }
+                                        }
```

Comment:
Do we need to throw exception here if `reader.TokenType != JsonTokenType.StartObject` while parsing variant objects?

Reply:
Yes, good catch thanks